src/wallet/orchard.h

```diff
@@ -0,0 +1,179 @@
+                tx.GetHash().begin(),
+                tx.GetOrchardBundle().inner.get());
+    }
+
+    /**
+     * Append each Orchard note commitment from the specified block to the
+     * wallet's note commitment tree.
+     *
+     * Returns `false` if the caller attempts to insert a block out-of-order.
+     */
+    bool AppendNoteCommitments(const int nBlockHeight, const CBlock& block) {
+        for (int txidx = 0; txidx < block.vtx.size(); txidx++) {
+            const CTransaction& tx = block.vtx[txidx];
+            if (!orchard_wallet_append_bundle_commitments(
+                    inner.get(),
+                    (uint32_t) nBlockHeight,
+                    txidx,
+                    tx.GetHash().begin(),
+                    tx.GetOrchardBundle().inner.get()
+                    )) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    bool TxContainsMyNotes(const uint256& txid) {
+        return orchard_wallet_tx_contains_my_notes(
+                inner.get(),
+                txid.begin());
+    }
+
+    void AddSpendingKey(const libzcash::OrchardSpendingKey& sk) {
+        orchard_wallet_add_spending_key(inner.get(), sk.inner.get());
+    }
+
+    void AddFullViewingKey(const libzcash::OrchardFullViewingKey& fvk) {
+        orchard_wallet_add_full_viewing_key(inner.get(), fvk.inner.get());
+    }
+
+    std::optional<libzcash::OrchardIncomingViewingKey> GetIncomingViewingKeyForAddress(
+            const libzcash::OrchardRawAddress& addr) const {
+        auto ivkPtr = orchard_wallet_get_ivk_for_address(inner.get(), addr.inner.get());
+        if (ivkPtr == nullptr) return std::nullopt;
+        return libzcash::OrchardIncomingViewingKey(ivkPtr);
+    }
+
+    /**
+     * Adds an address/IVK pair to the wallet, and returns `true` if the
+     * IVK corresponds to a full viewing key known to the wallet, `false`
+     * otherwise.
+     */
+    bool AddRawAddress(
+            const libzcash::OrchardRawAddress& addr,
+            const libzcash::OrchardIncomingViewingKey& ivk) {
+        return orchard_wallet_add_raw_address(inner.get(), addr.inner.get(), ivk.inner.get());
+    }
+
+    static void PushOrchardNoteMeta(void* orchardNotesRet, RawOrchardNoteMetadata rawNoteMeta) {
+        uint256 txid;
+        std::move(std::begin(rawNoteMeta.txid), std::end(rawNoteMeta.txid), txid.begin());
+        OrchardOutPoint op(txid, rawNoteMeta.actionIdx);
+        // TODO: what's the efficient way to copy the memo in the OrchardNoteMetadata
+        // constructor?
+        std::array<uint8_t, ZC_MEMO_SIZE> memo;
+        std::move(std::begin(rawNoteMeta.memo), std::end(rawNoteMeta.memo), memo.begin());
+        OrchardNoteMetadata noteMeta(
+                op,
+                libzcash::OrchardRawAddress(rawNoteMeta.addr),
+                rawNoteMeta.noteValue,
+                memo);
+        // TODO: noteMeta.confirmations is only available from the C++ wallet
```

Comment:
Please file an issue to fix this.

Reply:
Added https://github.com/zcash/zcash/issues/5566